### ISSUE TITLE
Bar chart with one series displays broken if chartColors argument is given

### ISSUE THREAD

**Issue by andrei-cs** [gitbrent/PptxGenJS#285]
Creating a bar chart with one series and providing a chartColors array seems results in a graph looking like this:

<img width="747" alt="screen shot 2018-02-10 at 20 36 16" src="https://user-images.githubusercontent.com/3409418/36065375-79590d26-0ea2-11e8-8819-19f281b099ae.png">

Removing the chartColors param displays it correctly but I can't find another way of setting the color.

The problem is exactly as in this issue - https://github.com/gitbrent/PptxGenJS/issues/225

Providing a single element in the chartColors array  solves the color problem but still creates a legend with all the values in the labels array.

Looking through the code it seems like it is somewhat of an intended behaviour:

https://github.com/gitbrent/PptxGenJS/blob/e4c46021f91deb4b34be9043e72d5f5ae37ab391/dist/pptxgen.js#L2691-L2695

How would one set a custom color for a bar chart ?

**Comment by gitbrent** [comment]
Hi @andrei-cs,

I'm not quite sure what the chart you're trying to produce would look like, but `chartColors` is the way to pass colors.

Using an array of colors with a single array is a specialized feature, any "normal" chart would just use something like this:
```javascript
chartColors: ['0088CC', '99FFCC'],
```

![screen shot 2018-02-11 at 18 42 19](https://user-images.githubusercontent.com/7218970/36080777-55666d8c-0f5b-11e8-92ea-f80e1b46580f.png)

Have you looked through the chart code in `examples/pptxgenjs-demo.html` ?

**Comment by andrei-cs** [comment]
Maybe I wasn't clear enough explaining my problem.

The following code: 

```
var slide = pptx.addNewSlide();
	var opts = { x:1.0, y:1.0, font_size:42, color:'00FF00' };
 	var data = [ { name: 'Sample',
    labels: [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23 ],
    values: [ 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 2, 1, 0, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0 ] } 
];

  slide.addChart(pptx.charts.BAR, data, {showLegend:true});
	pptx.save(); 
```

generates the graph below with the correct legend being the Series name

<img width="580" alt="screen shot 2018-02-12 at 13 07 20" src="https://user-images.githubusercontent.com/3409418/36094200-f899295a-0ff5-11e8-8590-64147cb543b5.png">


Adding chartColors to it:

```
var pptx = new PptxGenJS();
	var slide = pptx.addNewSlide();
	var opts = { x:1.0, y:1.0, font_size:42, color:'00FF00' };
 	var data = [ { name: 'Sample',
    labels: [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23 ],
    values: [ 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 2, 1, 0, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0 ] } 
];

  slide.addChart(pptx.charts.BAR, data, {showLegend:true, chartColors:['0000FF']});
	pptx.save();
```

Generates it like in the picture below - with the legend not being correct

<img width="644" alt="screen shot 2018-02-12 at 13 07 02" src="https://user-images.githubusercontent.com/3409418/36094263-2cd2d446-0ff6-11e8-9a3d-f742c0a7d249.png">

**Comment by andrei-cs** [comment]
@gitbrent Have you had time to look this over ?

**Comment by RokasDie** [comment]
Any news on this issue? Seems like if you add chartColors array (for chart with single item) with single color, it creates 
![image](https://user-images.githubusercontent.com/44611740/52412146-95779e80-2ae6-11e9-9025-b81aa43cbc99.png)

Instead of this layout, with correct legend: 

![image](https://user-images.githubusercontent.com/44611740/52412056-6103e280-2ae6-11e9-967b-acd9f0074cb7.png)

Here is the options variable:

```
const dataChartColumn = [
    {
      name: "Single item",
      labels: labels,
      values: returns,
    }
  ];
  const multiOpts = {
    x: 0.472441, y: 1.61417, w: 5.826772, h: 1.318898,
    showPercent: false,
    chartColors: ['003755'],
    barDir: "col",
    
    shadow: "none",
    valAxisLabelFormatCode: '0%',
    showLegend: true,
    legendPos: "b",
    legendFontSize: 8,
    legendFontFace: "Danske Text",

    catAxisLabelFontSize: 8,
    catAxisLabelFontFace: 'Danske Text',
    valAxisLabelFontFace: 'Danske Text',
    valAxisLabelFontSize: '8',
    valGridLine: {
      color: "EFF0F1",
    }
  }

  slide.addChart(pptx.charts.BAR, dataChartColumn, multiOpts);
```

**Comment by gitbrent** [comment]
Not really.  I've looked through the code a few times and while i intuitively understand that the use of this color array must be creating new series or something, i'm not seeing it.

The Legend is a straight toggle in XML.  It's there or it isn't - i'm not adding "rows" or items to, therefore, the `chartColors` must be doing something to affect the data series...?

**Comment by Slidemagic** [comment]
I experienced the same issue: in my case I am not using a legend, but negative values in a bar or column chart product different colours.

One possible source of this could be in the chart styles that PowerPoint is using. Have a look at the attached screen shot. In the bar above, you see suggested layouts: some of them have different colours for negative values, others not.
<img width="1595" alt="Screenshot 2019-05-21 07 34 36" src="https://user-images.githubusercontent.com/5433253/58068733-33294200-7b9b-11e9-94e0-07d51467ceca.png">

**Comment by kupkama2** [comment]
Any update on this? I am unable to correctly display legend for barchart with just one series + set chartColors. If I don't set the chartColor legend displays correctly.

**Comment by kupkama2** [comment]
@gitbrent is anyone looking into this? It's exactly the same probel as solved in #225 
Thanks for your response.

**Comment by gitbrent** [comment]
I'll look into it for 3.1

**Comment by gitbrent** [comment]
Hi @andrei-cs 

What's happening is that PowerPoint creates this unique "1 legend item for every category" legend when you use different colors for the bars.

Here is PowerPoint 2019 creating a bar chart from scratch and adding distinct colors:

> Basic legend

<img width="1314" alt="Screen Shot 2020-01-17 at 23 30 04" src="https://user-images.githubusercontent.com/7218970/72659249-47b81700-3982-11ea-904b-68251ece340f.png">

> When using a custom bar color:

<img width="1301" alt="Screen Shot 2020-01-17 at 23 30 24" src="https://user-images.githubusercontent.com/7218970/72659250-4b4b9e00-3982-11ea-976c-2c797766cc08.png">

**Comment by gitbrent** [comment]
@Slidemagic - thanks for the details.  This issue is fixed in master; target v3.1.0

<img width="584" alt="Screen Shot 2020-01-17 at 23 53 29" src="https://user-images.githubusercontent.com/7218970/72659485-e8f49c80-3985-11ea-8c69-92f666796523.png">

**Comment by pinakipatrapro** [comment]
Same issue in 3.12.0. 
Workaround : 
`chartColors: chartColors.splice(0,data.length)`
